src/server/search/aggregator.cc

```diff
@@ -104,34 +115,42 @@ PipelineStep MakeGroupStep(absl::Span<const std::string_view> fields,
 }
 
 PipelineStep MakeSortStep(std::string_view field, bool descending) {
-  return [field = std::string(field), descending](std::vector<DocValues> values) -> PipelineResult {
+  return [field = std::string(field), descending](PipelineResult result) -> PipelineResult {
+    auto& values = result.values;
+
     std::sort(values.begin(), values.end(), [field](const DocValues& l, const DocValues& r) {
       auto it1 = l.find(field);
       auto it2 = r.find(field);
       return it1 == l.end() || (it2 != r.end() && it1->second < it2->second);
     });
-    if (descending)
+
+    if (descending) {
       std::reverse(values.begin(), values.end());
-    return values;
+    }
```

Comment:
instead of reverse, you can do correct sorting

Reply:
It was added by Vlad and is fixed in the [second PR](https://github.com/dragonflydb/dragonfly/pull/4232/files#diff-aa06be8c9cd1ea3b09119db79df35e7f31b52e92aa6a6a3fe1c5f83f4b9e0269R133). So I don't think we need to change it here